lib/cartopy/mpl/gridliner.py

```diff
@@ -90,13 +97,42 @@ def _north_south_formatted(latitude, num_format='g'):
                                            _north_south_formatted(v))
 
 
+def _text_angle_to_specs_(angle):
+    """Get appropriate kwargs for a rotated label from its angle in degrees"""
+    if False and matplotlib.__version__ > '3':
+        # deactivated because does not work properly
+        # with rotation_mode='anchor'
+        va_align_center = 'center_baseline'
+    else:
+        va_align_center = 'center'
+    angle %= 360
+    if angle > 180:
+        angle -= 360
+    kw = {'rotation': angle, 'rotation_mode': 'anchor'}
+    if abs(angle) <= 45:
+        kw.update(ha='left', va=va_align_center)
+        loc = 'right'
+    elif abs(angle) >= 135:
+        kw.update(ha='right', va=va_align_center)
+        kw['rotation'] -= np.sign(angle) * 180
+        loc = 'left'
+    elif angle > 45:
+        kw.update(ha='center', va='bottom', rotation=angle-90)
+        loc = 'top'
+    else:
+        kw.update(ha='center', va='top', rotation=angle+90)
+        loc = 'bottom'
+    return kw, loc
```

Comment:
I'd like to see some clearer separations in this block please.  I can't see anything wrong with it functionally, but it would be clearer if there was a blank line above each variable that is not set within an `if` statement.  It would also make the logic and flow more apparent.

Reply:
Fixed